onadata/apps/api/viewsets/merged_xform_viewset.py

```diff
@@ -70,8 +84,18 @@ def form(self, *args, **kwargs):
     def data(self, request, *args, **kwargs):
         """Return data from the merged xforms"""
         merged_xform = self.get_object()
+        export_type = self.kwargs.get("format", request.GET.get("format"))
         queryset = Instance.objects.filter(
-            xform__in=merged_xform.xforms.all()
+            xform__in=merged_xform.xforms.all(),
+            deleted_at__isnull=True
         ).order_by("pk")
 
+        if export_type == "geojson":
+            page = CountOverridablePageNumberPagination(
```

Comment:
instead of doing this you can define `pagination_class` within the class and set to `CountOverridablePageNumberPagination`

`pagination_class = CountOverridablePageNumberPagination`

then do

`page = self.paginate_queryset(queryset)`

Reply:
Unrelated tests seem to fail when I set a `pagination_class` that applies to the whole viewset